setupbase.py

```diff
@@ -444,56 +420,39 @@ def initialize_options(self):
     def finalize_options(self):
         self.force = bool(self.force)
 
-    apps = ['notebook', 'tree', 'edit', 'terminal', 'auth']
-    targets = [ pjoin(static, app, 'js', 'built', 'main.min.js') for app in apps ]
+    target = pjoin(static, 'built', 'index.js')
+    targets = [target]
     
-    def sources(self, name):
+    def sources(self):
         """Generator yielding .js sources that an application depends on"""
-        yield pjoin(static, name, 'js', 'built', 'main.min.js')
-
-        for sec in [name, 'base', 'auth']:
-            for f in glob(pjoin(static, sec, 'js', '*.js')):
-                if not f.endswith('.min.js'):
-                    yield f
-        yield pjoin(static, 'services', 'built', 'config.js')
-        yield pjoin(static, 'built', 'index.js')
-        if name == 'notebook':
-            for f in glob(pjoin(static, 'services', '*', '*.js')):
-                yield f
-        for parent, dirs, files in os.walk(pjoin(static, 'components')):
-            if os.path.basename(parent) == 'MathJax':
+        yield pjoin(repo_root, 'package.json')
+        yield pjoin(repo_root, 'webpack.config.js')
+        for parent, dirs, files in os.walk(static):
+            if os.path.basename(parent) in {'MathJax', 'built'}:
                 # don't look in MathJax, since it takes forever to walk it
+                # also don't look at build targets as sources
                 dirs[:] = []
                 continue
             for f in files:
+                if not f.endswith('.js'):
+                    continue
                 yield pjoin(parent, f)
     
-    def should_run(self, name, target):
-        if self.force or not os.path.exists(target):
+    def should_run(self):
+        if self.force or not os.path.exists(self.target):
+            print("Missing %s" % self.target)
             return True
-        target_mtime = mtime(target)
-        for source in self.sources(name):
+        target_mtime = mtime(self.target)
+        for source in self.sources():
             if mtime(source) > target_mtime:
-                print(source, target)
+                print('%s > %s' % (source, self.target))
```

Comment:
Nvm, just spotted that it will only print once, because of the short circuit return.